store/src/java/com/zimbra/cs/account/ldap/LdapProvisioning.java

```diff
@@ -10853,4 +10853,30 @@ public void refreshUserCredentials(Account account) throws ServiceException {
             LdapClient.closeContext(zlc);
         }
     }
+
+    @Override
+    public String createAddressList(Domain domain, String name, String desc, Map<String, Object> attrs) throws ServiceException {
+        String domainDn = ((LdapEntry)domain).getDN();
+        String dn = LdapConstants.ATTR_uid + "=" + name + "," + LdapConstants.ATTR_ou + "=" + LdapConstants.PEOPLE + "," + domainDn;
+        Set<String> oc = new HashSet<String>();
+        oc.add(AttributeClass.OC_zimbraAddressList);
+        ZMutableEntry entry = LdapClient.createMutableEntry();
+        if (attrs != null && attrs.size() > 0) {
+            entry.mapToAttrs(attrs);
+        }
+        entry.addAttr(A_objectClass, oc);
+        String zimbraIdStr = LdapUtil.generateUUID();
+        entry.setAttr(A_zimbraId, zimbraIdStr);
+        entry.setAttr(Provisioning.A_uid, name);
+        if (!StringUtil.isNullOrEmpty(desc)) {
+            entry.setAttr(Provisioning.A_description, desc);
+        }
+        entry.setAttr(Provisioning.A_zimbraIsAddressListActive, "TRUE");
+        entry.setDN(dn);
+
+        ZLdapContext zlc = LdapClient.getContext(LdapServerType.MASTER, LdapUsage.CREATE_ADDRESS_LIST);
```

Comment:
Close zlc in finally block
finally {
            if (zlc != null) {
                refreshEntry(entry, zlc);
            }
            if (initZlc == null) {
                LdapClient.closeContext(zlc);
            }
        }